src/operator/first.ts

```diff
@@ -5,14 +5,14 @@ import {tryCatch} from '../util/tryCatch';
 import {errorObject} from '../util/errorObject';
 import {EmptyError} from '../util/EmptyError';
 
-export function first<T, R>(predicate?: (value: T, index: number, source: Observable<T>) => boolean,
+export function first<T, R>(predicate?: (value: T, index: number, observable: Observable<T>) => boolean,
                             resultSelector?: (value: T, index: number) => R,
-                            defaultValue?: any): Observable<T> | Observable<R> {
+                            defaultValue?: R): Observable<R> {
```

Comment:
Sorry, seems I'm little bit confused.. My thought in here is return type could remain as-is `: Observable<T> | Observable<R>` since it's already satisfies noimplicitAny conditions, and if this signature change's required than could brought up with separate PR/issue - do I have some incorrect understandings?